ds-collector/ds-collector

```diff
@@ -255,7 +258,10 @@ bastion_checks() {
   [ "$use_docker" = "true" ] || [ "$use_k8s" = "true" ] || ! [ -n "$sshPassword" ] || command -v sshpass >/dev/null 2>&1 || { echo >&2 "sshpass needs to be installed"; exit 1; }
   ! [ "$use_docker" = "true" ] || command -v docker >/dev/null 2>&1 || { echo >&2 "docker needs to be installed"; exit 1; }
   ! [ "$use_k8s" = "true" ] || command -v kubectl >/dev/null 2>&1 || { echo >&2 "kubectl needs to be installed"; exit 1; }
-  [ $(df "$baseDir" | tail -n +2 | awk '{print $4}') -ge 1000000 ] ||  { echo >&2 "There must be at least 1GB free at $baseDir"; exit 1; }
+  # detect if df supports --portability
+  DF_OPT=""
```

Comment:
why aren't we doing this just once, towards the top of the script?

Reply:
avoiding "globals". there are different invocation paths here (on the bastion and on the node).

Reply:
sure but at the expense of less re-use and more complexity in the code. Could possibly wrap that in a function but that's a bit hack-ey as well... it's fine, and as you say there is some precedence here.

Reply:
yes, i think wrapper function approach like `_df()` and `_timeout()` would be a nice improvement.